src/utils.py

```diff
@@ -234,33 +224,120 @@ def generate_translated_speech(path_to_voice_samples: str, translation: List[Tra
+    wave, sr = librosa.load(target_voice_path)
+    original_duration = len(wave) / sr
+    out = my_tts.inference(
+        text=text,
+        target_voice_path=target_voice_path,
+        output_wav_file=output_wav_file,
+        speed=1
+    )
+    generated_duration = len(out) / 24_000
+    
+    out = my_tts.inference(
+        text=text,
+        target_voice_path=target_voice_path,
+        output_wav_file=output_wav_file,
+        speed=generated_duration/original_duration
+    )
 
-    # Iterate over the transcription elements to load each translated audio segment
-    for i, element in enumerate(transcription):
-        segment_path = os.path.join(path_to_generated_audio, f"translated_segment_{i + 1}.wav")
+    print("Generated duration: {:.2f}".format(len(out)/24_000))
 
-        # Check if the segment file exists
-        if not os.path.exists(segment_path):
-            print(f"Warning: Translated segment not found for segment {i + 1}. Skipping.")
-            continue
 
-        # Load the translated audio segment
+def merge_audio_samples(path_to_generated_audio: str,
+                        original_audio_path: str,
+                        translation: List[TranscriptionElement]) -> str:
+    """ Creates an audio file from translated speech to replace the original one """
+    wave, sr = librosa.load(original_audio_path)
+    original_duration = len(wave) / sr
+
+    _, translated_sample_rate = librosa.load(os.path.join(path_to_generated_audio, f"translated_segment_1.wav"))
+    result_audio = AudioSegment.silent(duration=1000*original_duration)
+
+    for i, element in enumerate(translation):
+        segment_path = os.path.join(path_to_generated_audio, f"translated_segment_{i + 1}.wav")
         segment_audio = AudioSegment.from_file(segment_path)
-        combined_audio += segment_audio
-        print(f"Merged segment {i + 1} [{element.time_start:.2f} --> {element.time_end:.2f}]")
+        position = int(1000 * element.time_start)
+        result_audio = result_audio.overlay(segment_audio, position=position)
 
-    # Define the output path for the final merged audio file
     final_output_path = "final_translated_audio.wav"
-    
-    # Export the combined audio to a WAV file
-    combined_audio.export(final_output_path, format="wav")
+    result_audio.export(final_output_path, format="wav")
     print(f"Final merged audio saved at {final_output_path}")
-
     return final_output_path
-    
+
+
+def merge_translation(translation: List[TranscriptionElement],
+                      max_duration: int = 20,
+                      max_chars = 350) -> List[TranscriptionElement]:
+    """ Creates longer phrases for better speech generation """
+    print("Merging translations")
+    merged_translation = []
+    current_element = translation[0]
+    for next_element in translation:
+        if current_element == next_element:
```

Comment:
for next_element in translation:  --> for next_element in translation[1:]:

remove if statement.